scripts/vaccination_alerts/parse-vaccination-phases.ts

```diff
@@ -0,0 +1,168 @@
+  descriptionUrl: string;
+  notes: string;
+}
+
+const parseRow = (row: { [key: string]: string }): CsvRow => {
+  return {
+    state: row['State'],
+    eligibilityUrl: row['Eligibility Information URL'],
+    currentlyEligible: row['Currently Eligible'].trimRight() !== '',
+    startDate: row['Estimated Start Date'],
+    phase: row['Phase'],
+    tier: row['Tier'],
+    groupDetails: row['Detailed Group'],
+    generalGroup: row['General Group'],
+    descriptionUrl: row['Expanded description url'],
+    notes: row['Notes for Internal QA'],
+  };
+};
+
+const buildRegionGroup = (
+  state: string,
+  rows: CsvRow[],
+): RegionVaccinePhaseInfo => {
+  const firstRow = rows[0];
+
+  return {
+    locationName: state,
+    fips: regions.findByFullName(state.trimRight())!.fipsCode,
+    eligibilityInfoUrl: firstRow.eligibilityUrl,
+    emailAlertVersion: 1,
+    phaseGroups: rows.map(row => ({
+      phase: row.phase,
+      tier: row.tier,
+      currentlyEligible: row.currentlyEligible,
+      description: row.generalGroup,
+      expandedDefinitionUrl: row.descriptionUrl,
+      startDate: row.startDate,
+      updatedAt: new Date().toISOString(),
+    })),
+  };
+};
+
+const updateStatePhases = (
+  cmsRecord: RegionVaccinePhaseInfo,
+  spreadsheetRecord: RegionVaccinePhaseInfo,
+): RegionVaccinePhaseInfo => {
+  return {
+    ...cmsRecord,
+    ...{
+      eligibilityInfoUrl: spreadsheetRecord.eligibilityInfoUrl,
+      phaseGroups: spreadsheetRecord.phaseGroups,
+    },
+  };
+};
+
+async function main(dryRun: boolean) {
+  const csv = await fetch(GOOGLE_SHEETS_CSV_URL);
+  const content = await csv.text();
+  const rows = parse(content, { columns: true });
+  const records = rows.map(parseRow);
+  const recordsByState = groupBy(records, value => value.state);
+
+  const existingFips = stateVaccinationPhases.map(phase => phase.fips);
+
+  const groups = Object.keys(recordsByState).map(key => {
+    return buildRegionGroup(key, recordsByState[key]);
+  });
+
+  const spreadsheetFips = groups.map(group => group.fips);
+
+  const matchingCMSRecords = stateVaccinationPhases.filter(phase =>
+    spreadsheetFips.includes(phase.fips),
+  );
+
+  const newStates = groups.filter(group => !existingFips.includes(group.fips));
+
+  const inCmsNotSpreadsheet = stateVaccinationPhases.filter(
+    phase => !spreadsheetFips.includes(phase.fips),
+  );
+
+  const updatedMatchingRecords = matchingCMSRecords.map(cmsRecord =>
+    updateStatePhases(
+      cmsRecord,
+      groups.find(group => group.fips === cmsRecord.fips)!,
+    ),
+  );
+
+  console.log(
+    `In spreadsheet but not CMS: ${newStates.map(state => state.locationName)}`,
+  );
+  console.log(
+    `In both: ${matchingCMSRecords.map(state => state.locationName)}`,
+  );
+
+  console.log(
+    `In CMS but not spreadsheet: ${inCmsNotSpreadsheet.map(
+      state => state.locationName,
+    )}`,
+  );
+
+  if (dryRun) {
+    console.log('Dry run -- exiting');
+    return;
+  }
```

Comment:
I'm not sure if we need the dry run mode, the user can always discard changes to the CMS locally before committing them